lib/puppet/defaults.rb

```diff
@@ -330,6 +336,27 @@ def self.default_diffargs
         improvement for features that are checked frequently.
       EOT
     },
+    :always_cache_misses => {
```

Comment:
That sounds like a good place to start.

(And I totally agree about cache vs work, I just couldn't think of a concise setting name that encapsulated it.)